pkg/syncer/endpoints/endpoint_downstream_controller.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	ddsif "github.com/kcp-dev/kcp/pkg/informer"
+	"github.com/kcp-dev/kcp/pkg/logging"
+)
+
+const (
+	ControllerName = "syncer-endpoint-controller"
+)
+
+func NewEndpointController(
+	syncerLogger logr.Logger,
+	downstreamClient dynamic.Interface,
+	ddsifForDownstream *ddsif.GenericDiscoveringDynamicSharedInformerFactory[cache.SharedIndexInformer, cache.GenericLister, informers.GenericInformer],
+	syncedInformers map[schema.GroupVersionResource]cache.SharedIndexInformer,
```

Comment:
why pass in a map if you can pass in `syncedInformers[endpointsGVR]` directly? The map it outer plumbing. Irrelevant here.

Reply:
This map is not necessary anymore with the change done to the ControllerManager (following your comment here) : the ControllerManager now only manages the start/stop of the GVR-specific controllers, not their creation anymore.

Reply:
> the ControllerManager now only manages the start/stop of the GVR-specific controllers, not their creation anymore.

well, not really in fact. See [other comment](https://github.com/kcp-dev/kcp/pull/2452#discussion_r1063720309).
However I removed this map.